pkg/templates/git.go

```diff
@@ -214,6 +215,10 @@ func syncTemplateFromRef(
 		source += "//" + repo.SubPath
 	}
 
+	if plumbing.IsHash(ref) {
+		ref = ref[:7]
```

Comment:
make sure it does not happen out-of-range.

Reply:
Try to get hash of reference from repo, and trim it as template version.